src/CMakeLists.txt

```diff
@@ -1,7 +1,7 @@
 #Copyright(c) Microsoft Corporation.All rights reserved.
 #Licensed under the MIT                        license.
 
-set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
any comments to move to 20? we are using 20 internally

Reply:
I am open to do the same. @harsha-simhadri what do you think ? std:14 was lacking a lot of things, specifically **std::any** but if the other stakeholders are fine with 20. It should be fine

Reply:
I am OK. @theantony @PhilipBAdams @yiylin  @Sanhaoji2 @gopalrs @daxpryce  opinions?

Reply:
I prefer to use 20 if no one has concerns

Reply:
It's fine for things within DiskANN, but it requires all consumers to enable C++20. AFAIK that's not the case everywhere (e.g. recently we wanted to use `std::span` in a codebase that uses DiskANN, but weren't able to since the codebase didn't enable the `/std:c++20` flag). I think C++17 should be OK.

Reply:
some consumers are not ready to use C++20. I am fine to use 17 now.